### PR TITLE
[P4-2288] Display categories on population pages

### PR DESCRIPTION
**Pull request by merlinc**
## Proposed changes

PMU users manage the population of establishments via category groups. This PR groups the establishments via these categories.

### What changed

- Categories have been added to the location model
- getPrisonFreeSpaces uses an include of `category`, and the response is grouped by category, with an added caption
- Population dashboard loops over many population tables 

### Issue tracking

- [P4-2288](https://dsdmoj.atlassian.net/browse/P4-2288)

## Screenshots

| Before | After |
| ------ | ----- |
| Screenshot_2020-12-15 Population for 14 to 20 Dec 2020 (This week) |Screenshot_2020-12-16 Population for 14 to 20 Dec 2020 (This week)|
| |Screenshot_2020-12-16 Population for 14 to 20 Dec 2020 (This week)(1)|

#### Automated testing

- [x] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [ ] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: test/fixtures/api-client/location.find.json [hunk 1]
```diff
@@ -17,7 +17,11 @@
           "created_at": "2019-10-09T11:10:53+01:00",
           "updated_at": "2019-10-09T11:10:53+01:00"
         }
-      ]
+      ],
+      "category": {
```
Comment (merlinc) [inline test/fixtures/api-client/location.find.json:21]: The categories in both this, and the `location.findAll.json` are synthetic, and are enough to pass the tests. 

Do they need to be more authentically recorded from staging?

File: common/lib/api-client/models/location.js [hunk 2]
```diff
@@ -10,6 +10,10 @@ module.exports = {
       jsonApi: 'hasMany',
       type: 'suppliers',
     },
+    category: {
```
Comment (merlinc) [inline common/lib/api-client/models/location.js:13]: `locations_free_spaces` returns a location object, so the category needs to be added onto the `location` model.

File: common/lib/api-client/models/index.js [hunk 3]
```diff
@@ -18,7 +18,6 @@ const gender = require('./gender')
 const image = require('./image')
 const journey = require('./journey')
 const location = require('./location')
-const locationsFreeSpaces = require('./locations-free-spaces')
```
Comment (merlinc) [inline common/lib/api-client/models/index.js:21]: The `locations-free-spaces` model is not actually used - as `location-free-spaces` service returns `location` models.

File: app/population/views/dashboard.njk [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
 {% block pageContent %}
-  {% if resultsAsPopulationTable | length  %}
+  {% if resultsAsPopulationTables | length  %}
     <div class="govuk-grid-row">
       <section class="govuk-grid-column-full govuk-!-margin-bottom-6">
-        {{ govukTable(resultsAsPopulationTable) }}
+      {% for table in resultsAsPopulationTables %}
+        {{ govukTable(table) }}
+      {% else %}
+        <h1>No population to display</h1>
```
Comment (merlinc) [inline app/population/views/dashboard.njk:10]: Handling a default, just in case there are no population tables for buggy reasons

File: test/fixtures/api-client/location.find.json (hunk 1 above)
Comment (solidgoldpig) [reply test/fixtures/api-client/location.find.json:21]: I don't think so

File: app/population/views/dashboard.njk (hunk 4 above)
Comment (solidgoldpig) [reply app/population/views/dashboard.njk:10]: - `No population to display` -->  locales file
- Should the string be wrapped in an `h1` element? The page will already have a heading defined in the `pageHeader` block.
- (NB. ignore this if this is Andy'd design...) any reason not to reuse the `appMessage` as in `common/templates/collection-as-table.njk`

eg.
Single_requests_sent_14_to_20_Dec_2020__This_week_

**Review by solidgoldpig — APPROVED** [review]

File: app/population/views/dashboard.njk (hunk 4 above)
Comment (merlinc) [reply app/population/views/dashboard.njk:10]: @solidgoldpig Those are all good suggestions - I'll get them added in :)

No population to display is an edge case the design doesn't cover.

File: test/fixtures/api-client/location.find.json (hunk 1 above)
Comment (merlinc) [reply test/fixtures/api-client/location.find.json:21]: If we decide later that they should be, we can re-record from staging.

File: app/population/views/dashboard.njk (hunk 4 above)
Comment (merlinc) [reply app/population/views/dashboard.njk]: Added in appMessage and locales as per your suggestion - thanks :smile:
